src/route.c

```diff
@@ -165,7 +165,7 @@ static struct route_entry *route_local_lookup(uint32_t dest, const struct netif_
     struct route_entry *route_node;
     hashkey = route_local_hashkey(dest, port);
     list_for_each_entry(route_node, &this_local_route_table[hashkey], list){
-        if (dest == route_node->dest.s_addr) {
+        if ((dest == route_node->dest.s_addr) && (port->id == route_node->port->id)) {
```

Comment:
Add more  check just in case `port` is NULL.
`if ((dest == route_node->dest.s_addr) && (port && port->id == route_node->port->id)) {`